blog/content/post/2022-03-15-fobnail_3rd_phase.md

```diff
@@ -0,0 +1,202 @@
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/). Also, make sure to
+read other posts related to this project by visiting
+[fobnail](https://www.blog.3mdeb.com/tags/fobnail/) tag.
+
+# Keys used in communication between Fobnail Token and attester
+
+In current phase two key pairs are used during communication, those are AIK and
+EK. Both belong to TPM installed on attester and their private keys never have
+to leave TPM (except API for creating AIK always returns it).
+
+### AIK
+
+Attester Identification Key is used to sign data originating from attester. It
+is created by TPM under Endorsement Hierarchy from random data, meaning it will
+be different with each creation. To keep it always the same we are making it
+persistent by saving it in NVRAM under index `0x8100F0BA` (`81` is constant
+telling this is persistent handle, `F0B` - short for Fobnail, `A` - AIK).
+
+After key is made persistent, there is no way of reading its private part. It
+can still be used for signing (and decryption, if key type allows it), but only
+if this operation is carried out by TPM.
+
+### EK
+
+Endorsement Key is also created by TPM, but instead of using random data as an
+input, it uses EPS (Endorsement Primary Seed) that is burnt into TPM during
+manufacturing. Because manufacturer knows this seed, it can also create EK
+certificate. Unless EPS is changed, EK will always be the same, but because its
+creation takes significant amount of time it is also made persistent under
+handle `0x8100F0BE`.
+
+In most cases RSA EK certificate can be read from TPM NVRAM (index `0x01C00002`)
+but sometimes it is absent (fTPM, simulated TPM). Because this certificate is
+used by Fobnail Token for verification against trusted CA certificate chains,
+we've created [a script for "manufacturing" it](https://github.com/fobnail/fobnail-attester/blob/main/tools/tpm_manufacture.sh)
+whenever it is absent.
+
+The only use of EK right now is proving that AIK comes from the same TPM.
+
+# Tying AIK with EK
+
+TPM has to have access to EK in order to load and use AIK, but this doesn't
+imply that external party can test that AIK lives indeed inside TPM, or if it
+was created outside TPM by man in the middle. EK is not a signing key, so it
+can't sign AIK and prove its validity that way. `TPM2_ActivateCredential()` call
+was made to prove association of loaded object (like a key pair) to a credential
+(like Storage Key, e.g. EK).
+
+To prove that AIK comes from the same as EK, Fobnail first obtains and verifies
+EK certificate and asks for public AIK in TSS format. Based on this data it
+creates encrypted blob of randomly created secret, which can be only decrypted
+on TPM that has both EK and AIK currently loaded, and AIK is under the same
+hierarchy as EK. By returning correct secret to Fobnail, attester proves this is
+the case.
```

Comment:
We can mention here that credential activation not only proves that AIK comes from TPM but also proves that it has proper flags set, like protection policy, and other flags that make this key and AIK and not just some ordinary key.

Reply:
Done